crates/core/tedge_api/src/entity_store.rs

```diff
@@ -874,6 +994,18 @@ impl EntityRegistrationMessage {
             None
         };
 
+        let health_service = if let Some(value) = properties.remove("@health") {
+            let JsonValue::String(value_str) = value else {
+                return None;
+            };
+            let Ok(health_service) = value_str.parse() else {
+                return None;
+            };
+            Some(health_service)
+        } else {
+            None
+        };
+
```

Comment:
I'll definitely spend some time in a follow-up PR to simplify this entity registration parsing code by using serde directly.